### ISSUE TITLE
pyright: ignore directive does not actually suppress diagnostic

### ISSUE THREAD

**Issue by raxod502** [microsoft/pyright#7843]
**Describe the bug**
The `pyright: ignore` directive does not seem to behave as documented.

**Code or Screenshots**

```python
import flask


class Server:
    def __init__(self):
        self.app = flask.Flask(__name__)
        self.setup_routes()

    def setup_routes(self):
        @self.app.route("/")
        def _route_app():  # pyright: ignore
            return "Hello world"


server = Server()
flask_app = server.app
```

This is with Flask 2.3.2.

Despite the presence of `pyright: ignore`, the line `def _route_app()` is flagged with the diagnostic `"_route_app" is not accessed`. I know the diagnostic is coming from Pyright because I can see it in the LSP log:

```
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 10,
          "character": 12
        },
        "end": {
          "line": 10,
          "character": 22
        }
      },
      "message": "\"_route_app\" is not accessed",
      "severity": 4,
      "source": "Pyright",
      "tags": [
        1
      ]
    }
  ]
```

A screenshot:

![image](https://github.com/microsoft/pyright/assets/6559064/ea70c721-0ac3-4a13-9363-401bb9d234c1)

The documentation at https://microsoft.github.io/pyright/#/comments?id=line-level-diagnostic-suppression says that the presence of the `pyright: ignore` comment should prevent any diagnostic from being emitted by Pyright on the line.

In addition, the diagnostic report seems incorrect in the first place because the variable is prefixed with an underscore, which makes it clear that the lack of use is intentional.

**VS Code extension or command-line**
This is using Pyright via Emacs `lsp-mode`. I ensured that the latest release Pyright 1.1.361 was installed as well as the latest `lsp-mode` b27d9db1e of 2024-05-03. Emacs 28.1, Python 3.10.12, Ubuntu 22.04.

**Comment by erictraut** [comment]
Pyright is working as designed here, so I don't consider this a bug.

The `# pyright: ignore` suppresses diagnostics like errors, warnings, and the like. It doesn't suppress tagged hints. Tagged hints are not intended to be displayed as regular diagnostics by a client. They are intended to be displayed in a more subtle way, such as grayed text. Some clients unfortunately treat tagged hints like regular diagnostics. If you're using such a client, you can tell pyright not to emit any tagged hints through the use of the "[pyright.disableTaggedHints](https://microsoft.github.io/pyright/#/settings?id=pyright-settings)" setting.

**Comment by raxod502** [comment]
I see. Now that I know the word to look for, I see previous discussion at https://github.com/microsoft/pyright/issues/2672. Thanks.
